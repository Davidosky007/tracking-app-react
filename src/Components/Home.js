import React, { Component } from 'react';
import Registration from './auth/Registration';

export default class Home extends Component {
  
  render() {
    return (
      <div>
        <h1>Home</h1>
        <h1>Home</h1>
        <Registration />
      </div>
    )
  }
}
