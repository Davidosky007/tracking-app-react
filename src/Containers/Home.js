import { useEffect } from 'react';
import PropTypes from 'prop-types';
import { fetchAllMeasurements } from '../logic/api';
import { loadToken } from '../logic/storage';
import MainInfo from '../Components/MainInfo';
import Nutrient from '../Components/Nutrients';
import newDate, { getMonths } from '../logic/months';
import { getMeasurementsByDate, total, last } from '../logic/measurements';

const Home = (props) => {
  const {
    measurements, getAllMeasurements, date, changeDate, //eslint-disable-line
  } = props;
  const months = getMonths(0);

  useEffect(() => {
    const token = loadToken();
    fetchAllMeasurements(token).then((measurements) => {
      getAllMeasurements(measurements);
    });

    const today = new Date();
    const day = today.getDate();
    const month = today.getMonth();
    const year = today.getFullYear();
    changeDate({ day, month, year });
  }, []);

  const handleDateChange = (dir) => {
    const [day, month, year] = newDate(dir, date.day, date.month, date.year);
    changeDate({ day, month, year });
  };

  const [mainInfo, list] = Object.keys(measurements).reduce(([m, l], key) => {
    const main = ['Weight', 'Energy', 'Energy burned'].includes(key);
    const result = main
      ? [[...m, { title: key, measurements: measurements[key] }], l]
      : [m, [...l, { title: key, measurements: measurements[key] }]];
    return result;
  }, [[], []]);

  const main = mainInfo.map((elem) => {
    const reduceMethod = elem.title === 'Weight' ? last : total;
    return (
      <MainInfo
        key={elem.title}
        unit={elem.title}
        measurements={elem.measurements}
        selectedDate={date}
        getMeasurementsByDate={getMeasurementsByDate}
        reduceMethod={reduceMethod}
      />
    );
  });

  const nutrient = list.map((elem) => (
    <Nutrient
      key={elem.title}
      unit={elem.title}
      measurements={elem.measurements}
      selectedDate={date}
      getMeasurementsByDate={getMeasurementsByDate}
      reduceMethod={total}
    />
  ));

  return (
    <div>
      <div className="flex space-between date">
        <button className="color-gray" type="button" onClick={() => handleDateChange(-1)}>&lt;</button>
        <p className="color-gray">{`${date.day}-${months[date.month].full}-${date.year}`}</p>
        <button className="color-gray" type="button" onClick={() => handleDateChange(1)}>&gt;</button>
      </div>
      <div className="p-l-20 p-r-20">
        <div className="info-wrapper m-t-30 flex">
          {main}
        </div>
        <div className="nutrient-wrapper m-t-30 flex wrap center">
          {nutrient}
        </div>
      </div>
    </div>
  );
};

Home.propTypes = {
  measurements: PropTypes.shape({}).isRequired,
  getAllMeasurements: PropTypes.func.isRequired,
  date: PropTypes.shape({
    day: PropTypes.number.isRequired,
    month: PropTypes.number.isRequired,
    year: PropTypes.number.isRequired,
  }).isRequired,
  changeDate: PropTypes.func.isRequired,
};

export default Home;
