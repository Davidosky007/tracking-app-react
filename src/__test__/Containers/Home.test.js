import renderer from 'react-test-renderer';
import Home from '../../Containers/Home';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home
      measurements={{}}
      getAllMeasurements={() => {}}
      date={{
        day: 1,
        month: 1,
        year: 1,
      }}
      changeDate={() => {}}
    />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
