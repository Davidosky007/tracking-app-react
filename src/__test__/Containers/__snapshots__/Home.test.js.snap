// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    className="flex space-between date"
  >
    <button
      className="color-gray"
      onClick={[Function]}
      type="button"
    >
      &lt;
    </button>
    <p
      className="color-gray"
    >
      1-February-1
    </p>
    <button
      className="color-gray"
      onClick={[Function]}
      type="button"
    >
      &gt;
    </button>
  </div>
  <div
    className="p-l-20 p-r-20"
  >
    <div
      className="info-wrapper m-t-30 flex"
    />
    <div
      className="nutrient-wrapper m-t-30 flex wrap center"
    />
  </div>
</div>
`;
